new-packages/ts-project/__tests__/state.test.ts

```diff
@@ -0,0 +1,695 @@
+import { expect, test } from 'vitest';
+import { createTestProject, testdir, ts } from './utils';
+
+test('should extract states recursively', async () => {
```

Comment:
imho a test file shouldn't start with something as "complex" as a "should extract states recursively". We need simpler tests first 😉